packages/components/src/components/Tooltip/TooltipFloatingUi.tsx

```diff
@@ -0,0 +1,213 @@
+    });
+
+    const { context } = data;
+
+    const hover = useHover(context, {
+        move: false,
+        enabled: controlledOpen == null,
+        delay,
+    });
+    const focus = useFocus(context, {
+        enabled: controlledOpen == null,
+    });
+    const dismiss = useDismiss(context);
+    const role = useRole(context, { role: 'tooltip' });
+
+    const interactions = useInteractions([hover, focus, dismiss, role]);
+
+    return useMemo(
+        () => ({
+            open,
+            setOpen,
+            ...interactions,
+            ...data,
+            arrowRef,
+        }),
+        [open, setOpen, interactions, data],
+    );
+};
+
+type ContextType = ReturnType<typeof useTooltip>;
+
+export const TooltipContext = createContext<ContextType | null>(null);
+
+export const useTooltipState = (): ContextType => {
+    const context = useContext(TooltipContext);
+
+    if (context == null) {
+        throw new Error('Tooltip components must be wrapped in <Tooltip />');
+    }
+
+    return context;
+};
+
+type TooltipFloatingUiProps = { children: ReactNode } & TooltipOptions;
+
+export const TooltipFloatingUi = ({ children, ...options }: TooltipFloatingUiProps) => {
+    // This can accept any props as options, e.g. `placement`,
+    // or other positioning options.
+    const tooltip = useTooltip(options);
+
+    return <TooltipContext.Provider value={tooltip}>{children}</TooltipContext.Provider>;
+};
+
+type TooltipTriggerProps = HTMLProps<HTMLElement>;
+
+export const TooltipTrigger = forwardRef<HTMLElement, TooltipTriggerProps>(
+    ({ children, ...props }, propRef) => {
+        const state = useTooltipState();
+
+        const childrenRef = (children as any).ref;
+        const ref = useMergeRefs([state.refs.setReference, propRef, childrenRef]);
+
+        if (!isValidElement(children)) {
+            return <div>Invalid React Element</div>;
+        }
+
+        return cloneElement(
+            children,
+            state.getReferenceProps({
+                ref,
+                ...props,
+                ...children.props,
+                'data-state': state.open ? 'open' : 'closed',
+            }),
+        );
+    },
+);
+
+export type ArrowProps = { ref: ArrowRef; context: UseFloatingReturn['context'] };
+
+type TooltipContentProps = HTMLProps<HTMLDivElement> & {
+    arrowRender?: (props: ArrowProps) => ReactNode;
+    appendTo?: HTMLElement | null | MutableRefObject<HTMLElement | null>;
+};
+
+export const TooltipContent = forwardRef<HTMLDivElement, TooltipContentProps>((props, propRef) => {
+    const { arrowRender, appendTo, ...htmlProps } = props;
+
+    const state = useTooltipState();
+    const { isInstantPhase, currentId } = useDelayGroupContext();
+    const ref = useMergeRefs([state.refs.setFloating, propRef]);
+
+    useDelayGroup(state.context, { id: state.context.floatingId });
+
+    const instantDuration = 0;
+    const duration = 250;
```

Comment:
I would prefer to define it on the top of the file and with upper case `const DURATION = 250`

Reply:
fixed in cf7c9c315c